modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/user/store/v1/UserStoreFailureTest.java

```diff
@@ -0,0 +1,124 @@
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+import org.wso2.carbon.automation.engine.context.TestUserMode;
+
+import java.io.IOException;
+
+public class UserStoreFailureTest extends UserStoreTestBase {
+
+    private static final String INCORRECT_DOMAIN_ID = "SkRCQy0x";
+    private static final String INCORRECT_TYPE_ID = "WSDFGHBCVNMLKI";
+
+    @Factory(dataProvider = "restAPIUserConfigProvider")
+    public UserStoreFailureTest(TestUserMode userMode) throws Exception {
+
+        super.init(userMode);
+        this.context = isServer;
+        this.authenticatingUserName = context.getContextTenant().getTenantAdmin().getUserName();
+        this.authenticatingCredential = context.getContextTenant().getTenantAdmin().getPassword();
+        this.tenant = context.getContextTenant().getDomain();
+    }
+
+    @BeforeClass(alwaysRun = true)
+    public void init() throws Exception {
+
+        super.testInit(API_VERSION, swaggerDefinition, tenant);
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void testConclude() {
+
+        super.conclude();
+    }
+
+    @BeforeMethod(alwaysRun = true)
+    public void testInit() {
+
+        RestAssured.basePath = basePath;
+    }
+
+    @AfterMethod(alwaysRun = true)
+    public void testFinish() {
+
+        RestAssured.basePath = StringUtils.EMPTY;
+    }
+
+    @DataProvider(name = "restAPIUserConfigProvider")
+    public static Object[][] restAPIUserConfigProvider() {
+
+        return new Object[][]{
+                {TestUserMode.SUPER_TENANT_ADMIN},
+                {TestUserMode.TENANT_ADMIN}
+        };
+    }
+
+    @Test
+    public void testAddUserStoreNegativeCase() throws Exception {
+
+        String body = readResource("add-secondary-user-store-negative.json");
+        Response response = getResponseOfPost(USER_STORE_PATH_COMPONENT, body);
+        validateErrorResponse(response, HttpStatus.SC_BAD_REQUEST, "SUS-65001");
+    }
+
+    @Test
+    public void testGetUserStoreByDomainIdNegativeCase() {
+
+        Response response = getResponseOfGet(USER_STORE_PATH_COMPONENT + PATH_SEPARATOR +
+                INCORRECT_DOMAIN_ID);
+        validateErrorResponse(response, HttpStatus.SC_NOT_FOUND, "SUS-60003");
+    }
+
+    @Test
+    public void testUpdateUserStoreByIdNegativeCase() throws IOException {
+
+        String body = readResource("user-store-add-secondary-user-store.json");
+        Response response = getResponseOfPut(USER_STORE_PATH_COMPONENT + PATH_SEPARATOR +
+                INCORRECT_DOMAIN_ID, body);
+        validateErrorResponse(response, HttpStatus.SC_NOT_FOUND, "SUS-60001");
+    }
+
+    @Test
+    public void testGetUserStoreTypeByIdNegativeCase() {
+
+        Response response = getResponseOfGet(USER_STORE_PATH_COMPONENT + USER_STORE_META_COMPONENT
+                + PATH_SEPARATOR + INCORRECT_TYPE_ID);
+        response.then().log().ifValidationFails().assertThat().statusCode(HttpStatus.SC_OK)
```

Comment:
if the id is not correct, it should return not found

Reply:
Fix with b68f8e3b41be55fe2c21fca74b10ce550402038a